### PR TITLE
[Refactor] Move some server and manager to GlobalStateMgr

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## Why I'm doing:
In the original Mgr class, the calling locations are hashed in many locations, which is not conducive to expansion and secondary development. leading to a potential violation of dependency inversion principle by instantiating concrete classes outside of the constructor. This PR moves the creation of multiple classes to GlobalStateMgr for unified creation and management.
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by imay** [comment]
can you explain why you did this?

**Comment by HangyuanLiu** [comment]
> can you explain why you did this?

In the original Mgr class, the calling locations are hashed in many locations, which is not conducive to expansion and secondary development. leading to a potential violation of dependency inversion principle by instantiating concrete classes outside of the constructor. This PR moves the creation of multiple classes to GlobalStateMgr for unified creation and management.

File: fe/fe-core/src/main/java/com/starrocks/leader/Checkpoint.java [hunk 1]
```diff
@@ -211,6 +212,7 @@ private boolean replayAndGenerateGlobalStateMgrImage(long checkPointVersion) {
         // generate new image file
         LOG.info("begin to generate new image: image.{}", checkPointVersion);
         globalStateMgr = GlobalStateMgr.getCurrentState();
+        globalStateMgr.setEditLog(new EditLog(null));
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/leader/Checkpoint.java:215]: Why use a null value to construct edit log?

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 2]
```diff
@@ -570,7 +588,12 @@ private GlobalStateMgr(boolean isCkptGlobalState, NodeMgr nodeMgr) {
         this.portConnectivityChecker = new PortConnectivityChecker();
 
         // Alter Job Manager
-        this.alterJobMgr = new AlterJobMgr();
+        // Alter Job Manager
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:591]: redundant

**Review by packy92 — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**